### PR TITLE
chore: update eslint spec rules

### PR DESCRIPTION
**Pull request by benjamincharity**
- use newer version of our eslint rules package
- disable rules for spec and mock files
- fix a few styling issues
- remove JS files from gitignore
- remove cli flag for eslint ignore in favor of eslint ignore file

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -3,28 +3,42 @@ module.exports = {
...
+      "env": {
+        "jest": true,
+      },
+      "rules": {
+        "dot-notation": "off",
+        "guard-for-in": "off",
+        "line-comment-position": "off",
+        "no-console": "off",
+        "no-magic-numbers": "off",
+        "no-underscore-dangle": "off",
```
Comment (atlwendy) [inline .eslintrc.js:40]: Finally! 👏

File: .gitignore [hunk 2]
```diff
@@ -5,8 +5,6 @@ node_modules/
 out-tsc/
 debug.log
 npm-debug.log
-src/**/*.js
-terminus-ui/**/*.js
 src/**/*.css
```
Comment (atlwendy) [inline .gitignore:8]: should we not ignore generated `js` files at all? (instead of having few exceptions)

**Review by atlwendy — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (benjamincharity) [reply .gitignore:8]: Isn’t that what this change does? Remove any rules to ignore js files?

Comment (atlwendy) [reply .gitignore:8]: Right... I just want to make sure we're intentional on watching ALL js files, including the ones that're generated from typescript..

**Pull request merged**